httputil/client.go

```diff
@@ -4,10 +4,19 @@ import (
 	"io"
 	"net/http"
 	"strings"
+	"time"
 )
 
 const (
 	headerXForwardedFor = "X-Forwarded-For"
+
+	// default transport settings
+	DefaultMaxIdleConnsPerHost = 10
+	DefaultIdleConnTimeout     = 30 * time.Second
```

Comment:
Can we go with the default go one here i.e. 90s? 30s seems a bit too short.
Even in the [aws sdk](https://github.com/rudderlabs/rudder-go-kit/blob/main/awsutil/awsutil.go#L98-L111) we're using 90s.

Reply:
Actually can we copy all settings from the `http.DefaultTransport` as a starting point? cc @lvrach